packages/parser/bin/index.ts

```diff
@@ -0,0 +1,33 @@
+import 'reflect-metadata';
+
+import { Amqp } from '@spectacles/brokers';
+import { AmqpResponseOptions } from '@spectacles/brokers/typings/src/Amqp';
+import { on } from 'events';
+import { Message } from '@spectacles/types';
+import * as postgres from 'postgres';
+import { Lexer, Parser, Tokens, Unordered } from 'lexure';
+
+const broker = new Amqp('gateway');
+const pg = postgres();
+
+(async () => {
+	await broker.connect('rabbitmq');
+	await broker.createQueue('COMMAND');
+	await broker.subscribe(['MESSAGE_CREATE']);
+	
+	for await (const [message, { ack }] of on(broker, 'MESSAGE_CREATE') as AsyncIterableIterator<[Message, AmqpResponseOptions]>) {
+		ack();
+		const [data] = await pg`select settings ->> 'PREFIX' as prefix
+			from settings
+			where guild_id = ${message.guild_id}
+				is not null;` as [{ prefix: string | null }];
+		const prefix = data.prefix ?? '?';
```

Comment:
Could `data` be undefined here if the guild has no settings? If so, this has the risk of crashing the parser.

Reply:
No, it would just be `{ prefix: null }`